src/utils/data_buffer.cpp

```diff
@@ -15,11 +29,14 @@ DataBuffer::~DataBuffer ()
 
 bool DataBuffer::is_ready ()
 {
-    return (data != NULL);
+    return (data != nullptr);
 }
 
 void DataBuffer::add_data (double *value)
 {
+    if (buffer_size < 2)
```

Comment:
Yes - this one was interesting. If you remove this check and use (for example) `DataBuffer buffer_zero (4, 0);`, this will generate an exception at the below (in `data_buffer.h`) when `add_data()` is invoked:

```c++
size_t next (size_t index)
{
    return (index + 1) % buffer_size;
}
```

If you do the same with `DataBuffer buffer_zero (4, 1);` it will work fine, but won't actually add any data to the buffer. I think this is confusing because when my buffer has size one, I would intuitively expect to be able to put one sample vector into it. The if statement avoids the exception in the case of a 0 sized buffer and saves a few cycles (same result) with a 1 sized buffer.

You can use the `AddData_BufferSizeIsLessThanTwo_DoNothing` test to verify.

Now that we're using `is_ready()` again, a better approach may be to do this check in the constructor and not mark the buffer as ready if it's size is less than 2.

Reply:
The other thing to note here is that the `DataBuffer` actually has capacity of size `buffer_size - 1` in practice. For example, if you have `buffer_size = 2` then you can add one data vector to the buffer (not 2).

Reply:
[89733f9](https://github.com/brainflow-dev/brainflow/pull/521/commits/89733f9e311563385b71fdd9b52cb567d57e882e) fixes this issue.

Reply:
I know that real size is -1, its pretty common implementation of ring buffer